src/views/home/index.js

```diff
@@ -14,9 +14,10 @@ import exitBlue from "../../utils/images/exit-blue.png";
 import close from "../../utils/images/close.png";
 import { withRouter } from "react-router-dom";
 import Wrapper from "../../components/wrapper";
+
 import config from "../../config/config.json";
 
-const dataSource = new DataSource();
+const dataSource = getDataSource();
```

Comment:
This should probably be set at state in react. While it doesn't seem to obvious why, it's good to maintain these "data sources" inside of React's lifecycle to avoid race conditions regarding data. While we can't do this with this class, we could for example hook in and listen to internal state changes and sync the below class accordingly. Hence, I think we should move it into state. Same with the other ones below!

Reply:
Agreed a 100%, some sort of provider is the way to go even. But I wasn't feeling like refactoring much. Let's evaluate the worth of doing this change at this stage for the explorer, if so. a `ProvideDataSource` component sounds like it

Reply:
Hmm, for the sake of complying to the react "ethos" we could make a context with this... but considering it's a class component, it might be a bit verbose yes. I'd say let's just make an issue for doing this later :)

Reply:
Yes sir